### ISSUE TITLE
Formatting Does Not Get Applied to Child Extras

### ISSUE THREAD

**Issue by JordanW9232** [mcstatus-io/mcutil#2]
I made an issue for this same problem a while ago in your old "minecraft-motd-util" repository here: https://github.com/PassTheMayo/minecraft-motd-util/issues/5. I made some changes that you merged that fixed it here: https://github.com/PassTheMayo/minecraft-motd-util/pull/4/files.

I have since moved to using your new mcstatus.io API, and I noticed this problem is back. Any styles on an "extra" object should get applied to all of its children, including grandchildren. It is currently not doing this.

I will use the same example I used in that issue. Here is what Complex Gaming's MOTD looks like: 
![221620263-6ae4770f-98c5-4c1f-a638-c44f51bf84ab](https://github.com/mcstatus-io/mcutil/assets/126441391/b2f9a549-8a5c-45ff-8e2f-c797cc45033d)

And here is what it looks like on your mcstatus.io site:
![image](https://github.com/mcstatus-io/mcutil/assets/126441391/cac4f05d-2c15-411e-a7a8-dab272d78439)

In this example the bold and obfuscated is not getting applied to the extra with color: aqua:
```
"extra":[
  {
    "bold":true,
    "obfuscated":true,
    "extra":[
      {
        "color":"light_purple",
        "extra":[
          {
            "color":"aqua",
            "text":"i"
          }
        ],
        "text":"i"
      }
    ],
    "text":"i"
  },
],
```

You can refer to the description of the issue in that other repository to understand why this isn't working. Maybe the changes in the pull request will help with the solution as well.

Sorry I'm not really familiar with Go to attempt this myself.

Edit:
I also noticed that the "light_purple" color is not being applied for some reason. I think it may be unrelated though since it looks like "light_purple" is set within the same extra object. Like here:
```
"color":"light_purple",
"extra":[
  {
    "color":"aqua",
    "text":"i"
  }
],
"text":"i"
```

The color isn't coming from a parent object so I'm not sure why the "i" text is not purple.

**Comment by JordanW9232** [comment]
Actually I took a stab at it even though I'm not familiar with Go. Seems to be working as it should!

Pull request here: https://github.com/mcstatus-io/mcutil/pull/3

Also, about the "light_purple" color not working that I mentioned, I realized that before I even made any changes your package already had the color applying as it should. So I'm not sure why it's not working when I use the API. Maybe the API is using an older version of mcutil?
